symengine/sets.h

```diff
@@ -199,5 +225,42 @@ inline RCP<const Set> interval(const RCP<const Number> &start,
         return finiteset({start});
     return emptyset();
 }
+
+// ! \return RCP<const Set>
+inline RCP<const Set> sym_union(const set_basic &in, bool solve = true)
```

Comment:
`solve` variable ensures that we don't fall in a loop when we have solved the `set_union` from respective containers and they are trying to return a `Union` object.